app/components/common/MapUK.component.js

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState, useCallback, useRef } from "react";
+import PropTypes from "prop-types";
+import {
+  Marker,
+  GoogleMap,
+  InfoWindow,
+  DirectionsRenderer,
+} from "@react-google-maps/api";
+import mapStyles from "./mapStyles";
+import { getInitials } from "./Avatar.component";
+
+const containerStyle = {
+  width: "100%",
+  height: "80vh",
+};
+
+const center = {
+  lat: 54.2511,
+  lng: -4.4632,
+};
+
+const options = {
+  styles: mapStyles,
+  disableDefaultUI: true,
+  zoomControl: true,
+};
+
+const MapUK = ({ teams }) => {
+  const London = new window.google.maps.LatLng(51.509865, -0.118092);
+  const Edinburgh = new window.google.maps.LatLng(55.953251, -3.188267);
+
+  const [selected, setSelected] = useState(null);
+  const [markers, setMarkers] = useState([]);
+  const [directions, setDirections] = useState([]);
+  const [setError] = useState(null);
+
+  useEffect(() => {
+    const google = window.google;
+
+    const directionsService = new window.google.maps.DirectionsService();
+    directionsService.route(
+      {
+        origin: London,
+        destination: Edinburgh,
+        travelMode: google.maps.TravelMode.DRIVING,
+      },
+      (result, status) => {
+        if (status === google.maps.DirectionsStatus.OK) {
+          setDirections(result);
+          let tempMarkers = [];
+          teams.map((team) => {
+            let distanceSum = 0;
+            for (let path of result.routes[0].overview_path) {
```

Comment:
On reflection I think it would be worth adding in the break clause for the edge case of when a team's totalDistance is greater than final point, whatever that action might be. You could achieve that through something like this, just need the index of current coordinate being checked.

`for (let [index, path] of result.routes[0].overview_path.entries()) {
              if (index >= result.routes[0].overview_path.length - 1) {
                console.log('Do something here')
                break;
              }`

Reply:
Further down there is a check that the teams distance has to be smaller or nothing will happen, only valid ones get added to the markers.

Reply:
The reason why it fails for team distance larger than the route is because in the `computeDistanceBetween()` the `nextPath` is accessed by adding +1 to the previous `path` index. Now when you get to the last one there is no next one, so the computeDistanceBetween needs LatLng object and tried to get it from an undefined point. This would not happen unless you are reaching the end of the route(or go beyond it). Solution: avoid trying to access that non existing point after the last one ie `if(!nextPath) break;`

Reply:
A nice solution!